utils/ansible-playbooks/deploy-monitoring/templates/ocp-monitor.j2

```diff
@@ -41,6 +41,64 @@ bash_backtrace() {
     echo "Exiting with status ${RET}"
     exit "${RET}"
 }
+
+## FUNCTIONS FOR STATE MACHINE CHECKS
+
+function get_current_state () {
+# Parameters passed:
+#  - unique name identifying what is being tracked state-wise
+#  - variable to hold the current state, either read from state file or initialized when not yet set.
+# Also forms the basis for the file-name used to track state.
+# If the file does not exist, then a state of 0 is assigned and the state file created.
+
+  local statename="$1"
+  local statevalue="$2"
+  local current_state=0
+
+  STATE_FILE="/tmp/ocp-monitor."$1".state"
+
+  if [ -f $STATE_FILE ]; then
+    current_state=`cat $STATE_FILE`;
+  else
+    current_state=`echo 0 | tee $STATE_FILE`
+  fi
+  eval $statevalue=$current_state
+}
+
+function set_state () {
+# Parameters passed:
+#  - unique name identifying what is being tracked state-wise
+#  - variable to hold the current state, either read from state file or initialized when not yet set.
+# Also forms the basis for the file-name used to track state.
+# If the file does not exist, then a state of 0 is assigned and the state file created.
+
+  STATE_FILE="/tmp/ocp-monitor."$1".state"
+  local evalstate = `echo $2 | tee $STATE_FILE`
+}
+
+## FUNCTIONS FOR NODE_READY CHECKS
+# These checks used to be placed in-line, but are now separated into separate functions so
+# as to increase clarity when adding the state checks for Node Ready health check.
+
+function test_if_all_nodes_are_ready () {
+# following test will return 1 if all nodes have a Ready state = True. Otherwise, return 0.
+
+  if oc get nodes -o go-template="{{range .items}}{{.metadata.name}}{{print \"\t\"}}{{range .status.conditions}}{{if eq .type \"Ready\"}}{{.status}}{{end}}{{end}}{{print \"\n\"}}{{end}}" | grep -v True
+  then
+    return 0
+  else
+    return 1
+  fi
+}
+
+function return_node_ready_count () {
+# counts and returns the number of nodes where Ready=True.
+# works around limits involving functions by passing by reference update to the calling parameter to the function.
+
+  local var1="$1"
```

Comment:
This is where the likes of BASH is showing its limitations. While the return_node_ready_count function is specialized and thus overkill, the functions get_current_state and set_state are meant to be able to be re-used within the script for other properties we may want to also track state for. 

The ```local VAR1="$1"``` is assigning the variable name passed to the function. The eval line evaluates the overall expression, since VAR1 has the name of the variable defined at the global level. 

I'll adjust the return_node_ready_count function since the current use is over-kill, but can't change how get_current_state and set_state work, since I want those functions to be able to be used as-is if/when we want to track other properties in a stateful manner. That or I can just switch to a language like Python.

If you have some other thoughts on how I can create re-usable functions without taking my approach, let me know.